tests/auto_run_tests.pl

```diff
@@ -2,60 +2,228 @@
+        "    -ExeSubDir <dir>         Subdirectory for finding the executables\n" .
+
+        "    --sandbox | -s <sandbox> Runs each program using a sandbox program\n" .
+        "    --dry-run | -z           Do everything except run the tests\n" .
+        "    --show-configs           Print possible values for -Config and -Excludes\n" .
+        "                             broken down by list file\n" .
+        "    --list-configs           Print combined set of the configs from the list\n" .
+        "                             files\n" .
+        "    --list-tests             List all the tests that would run\n" .
+        "    --stop-on-fail | -x      Stop on any failure\n";
+
+    exit(0);
+}
+
+# Parse Options
+my $help = 0;
+my $sandbox = '';
+my $show_configs = 0;
+my $list_configs = 0;
+my $list_tests = 0;
+my $stop_on_fail = 0;
+my $cmake = 0;
+my $ctest = 'ctest';
+my $ctest_args = '';
+my %opts = (
+    'help|h' => \$help,
+    'sandbox|s=s' => \$sandbox,
+    'dry-run|z' => \$dry_run,
+    'show-configs' => \$show_configs,
+    'list-configs' => \$list_configs,
+    'list-tests' => \$list_tests,
+    'stop-on-fail|x' => \$stop_on_fail,
+    'cmake' => \$cmake,
+    'ctest=s' => \$ctest,
+    'ctest-args=s' => \$ctest_args,
+);
+foreach my $list (@builtin_test_lists) {
+    if (!exists($list->{default})) {
+        $list->{default} = 0;
+    }
+    $list->{option} = undef;
+    my $opt = "$list->{name}!";
+    $opts{$opt} = \$list->{option};
+}
+Getopt::Long::Configure('bundling', 'no_auto_abbrev');
+if (!GetOptions(%opts)) {
+    print_usage(1);
+}
+elsif ($help) {
+    print_help();
 }
+for my $list (@builtin_test_lists) {
+    if (!exists($list->{enabled})) {
+        $list->{enabled} = defined($list->{option}) ? $list->{option} : $list->{default};
+    }
+}
+my $query = $show_configs || $list_configs || $list_tests;
 
-my @file_list;
+# Determine what test list files to use
+my @file_list = ();
+foreach my $list (@builtin_test_lists) {
+    push(@file_list, "$DDS_ROOT/$list->{file}") if ($query || $list->{enabled});
+}
+push(@file_list, @ARGV);
+foreach my $list (@file_list) {
+    die("$list is not a readable file!") if (!-r $list);
+}
 
-if ($opt_a || !scalar @opt_l) {
-    push(@file_list, "$DDS_ROOT/tests/dcps_tests.lst");
+if ($show_configs) {
+    foreach my $test_list (@file_list) {
+        my $config_list = new PerlACE::ConfigList;
+        $config_list->load($test_list);
+        print "$test_list: " . $config_list->list_configs() . "\n";
+    }
+    exit(0);
 }
 
-if (scalar @opt_l) {
-    push(@file_list, @opt_l);
+if ($list_configs) {
+    my %configs = ();
+    foreach my $test_list (@file_list) {
+        my $config_list = new PerlACE::ConfigList;
+        $config_list->load($test_list);
+        for my $config (split(/ /, $config_list->list_configs())) {
+            $configs{$config} = 1;
+        }
+    }
+    for my $config (sort(keys(%configs))) {
+        print("$config\n");
```

Comment:
```suggestion
        print "$config\n";
```
Just for consistency... there's a few more of these below as well